edweiss-app/__test__/utils/lectures/slides/utilsFunctions.test.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * @file utilsFunctions.test.tsx
+ * @description Unit tests for utilsFunctions.tsx to ensure correct
+ *              utils functionality and corner cases handling.
+ * @author Adamm Alaoui
+ */
+
+// ------------------------------------------------------------
+// --------------- Import Modules & Components ----------------
+// ------------------------------------------------------------
+
+import { langIconMap, langNameMap } from '@/utils/lectures/remotecontrol/utilsFunctions';
+import { transModeIconMap, transModeIDMap, transModeNameMap } from '@/utils/lectures/slides/utilsFunctions';
+
+
+// ------------------------------------------------------------
+// -----------------  Mocking dependencies    -----------------
+// ------------------------------------------------------------
+
+
+// Authentication mock
+jest.mock('@/contexts/auth', () => ({
+    useAuth: jest.fn(),
+}));
+
+// Sign in fucntions
+jest.mock('@/config/firebase', () => ({
+    signInWithGoogle: jest.fn(),
+    signInAnonymously: jest.fn(),
+    callFunction: jest.fn(),
+}));
+
+
+// Firebase modules
+jest.mock('@react-native-firebase/auth', () => ({
+    currentUser: { uid: 'anonymous' },
+    signInAnonymously: jest.fn(() => Promise.resolve({ user: { uid: 'anonymous' } })),
+}));
+
+
+// Firebase functions
+jest.mock('@react-native-firebase/functions', () => {
+    return {
+        httpsCallable: jest.fn(() => jest.fn(() => Promise.resolve({ data: { status: true } })))
+    };
+});
+
+// Storage module
+jest.mock('@react-native-firebase/storage', () => {
+    return {
+        ref: jest.fn(() => ({
+            putFile: jest.fn(() => Promise.resolve({})),
+            getDownloadURL: jest.fn(() => Promise.resolve('mocked_download_url')),
+            delete: jest.fn(() => Promise.resolve({}))
+        })),
+    };
+});
+
+
+// Firestore communication with database
+jest.mock('@react-native-firebase/firestore', () => ({
+    __esModule: true,
+    default: jest.fn(() => Promise.resolve({})),
+}));
+
+
+// ------------------------------------------------------------
+// ---------    ShowTime! Utils Functions Test suite    -------
+// ------------------------------------------------------------
+
+describe('Utils Function Testsuite', () => {
+    describe('transModeIconMap', () => {
+        it('should have correct icon for original mode', () => {
```

Comment:
Just a detail, but one of the feedbacks I received in the MS2 was that I could make a bit more comments to explain each of my tests. Maybe this could be interesting for you as well

Reply:
Yes you are right, I will adapt it right away.
Just for your information, when doing the test suites. I am trying to segmentate the describes into the different functions that it tackles, so that's it more modular and brings structure and flexibility to the future updates.

Reply:
Yes I noticed, but a little bit of comments as well won't hurt